frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -150,69 +178,89 @@ export default ({
       })
       let mt = `message-${index}-${num}`
       switch (message.from) {
-        case messageTypes.NOTIFICATION:
+        case MESSAGE_TYPES.NOTIFICATION:
           mt = `notification-${index}-${num}`
           break
 
-        case messageTypes.INTERACTIVE:
+        case MESSAGE_TYPES.INTERACTIVE:
           mt = `interactive-${index}-${num}`
           break
       }
       return mt
     },
     messageType (message) {
       let mt = 'message'
-      switch (message.from) {
-        case messageTypes.NOTIFICATION:
+      switch (message.type) {
+        case MESSAGE_TYPES.NOTIFICATION:
           mt += '-notification'
           break
 
-        case messageTypes.INTERACTIVE:
+        case MESSAGE_TYPES.INTERACTIVE:
```

Comment:
Here and above where there's also a `switch` statement, an object map could be used instead for better performance and more compact/cleaner code:

```js
return {
  [MESSAGE_TYPES.NOTIFICATION]: 'message-notification',
  [MESSAGE_TYPES.INTERACTIVE]: 'message-interactive',
}[message.type]
```

Please check the rest of the changes where such a change can be made to convert `switch` statements to object maps + lookups.

Reply:
Yeap. Make sense. Really nice solution...